sway-fmt-v2/src/utils/attributes.rs

```diff
@@ -1,11 +1,69 @@
 use crate::Formatter;
-use sway_parse::AttributeDecl;
+use sway_parse::attribute::AttributeDecl;
 use sway_types::Spanned;
-pub fn format_attributes(attributes: Vec<AttributeDecl>, _formatter: &mut Formatter) -> String {
-    // TODO format attributes
-    attributes
-        .into_iter()
-        .map(|x| x.span().as_str().to_string())
-        .collect::<Vec<String>>()
-        .join("\n")
+
+use super::bracket::{Parenthesis, SquareBracket};
+
+trait Format {
+    fn format(&self, line: &mut String, formatter: &mut Formatter);
+}
+
+pub fn format_attributes(attributes: Vec<AttributeDecl>, formatter: &mut Formatter) -> String {
+    let mut formatted_code = String::new();
+    for attr in &attributes {
+        AttributeDecl::format(attr, &mut formatted_code, formatter);
+    }
+
+    formatted_code
+}
+
+impl Format for AttributeDecl {
+    fn format(&self, line: &mut String, formatter: &mut Formatter) {
+        // At some point there will be enough attributes to warrant the need
+        // of formatting the list according to `config::lists::ListTactic`.
+        // For now the default implementation will be `Horizontal`.
+        //
+        // `#`
+        line.push_str(self.hash_token.span().as_str());
+        // `[`
+        Self::open_square_bracket(line, formatter);
+        let attr = self.attribute.clone().into_inner();
+        // name e.g. `storage`
+        line.push_str(attr.name.span().as_str());
+        // `(`
+        Self::open_parenthesis(line, formatter);
+        // format and add args `read, write`
+        if let Some(args) = attr.args {
+            let mut args = args
+                .into_inner()
+                .value_separator_pairs
+                .iter()
+                .map(|arg| format!("{}{}", arg.0.as_str(), arg.1.span().as_str()))
+                .collect::<Vec<String>>()
+                .join(" ");
+            args.pop(); // pop the ending space
+            args.pop(); // pop the ending comma
+            line.push_str(&args);
+        }
```

Comment:
Should we have something like `Self::close_parenthesis(line, formatter);` here?

Reply:
It is a bit needless, but I figured I'd use the traits for consistency